### DISCUSSION TITLE
Default Configuration - install_root_path

### DISCUSSION THREAD

**Discussion by syntaxman** [mason-org/mason.nvim#313] [Q&A] [answered]
I used the default configuration file for the purpose of learning about lua/lsp/mason.  I realize these are defaults I'd get if I didn't use the file at all.  However, I want to understand what I'm missing here.  

This line causes an error at startup:

install_root_dir = path.concat { vim.fn.stdpath "data", "mason" },

The error is:
[packer.nvim] [ERROR 20:57:28] packer.lua:983: Failure running setup function: "/home/wade/.config/nvim/lua/mason_cfg.lua:30: attempt to index global 'path' (a nil value)"

Someone in IRC mentioned plenary, so I included it and called plenary.path:concat.  (that may be ridiculous, but I was without much of a clue).  The path was recognized, but concat was not.

Would someone please help me figure out what it is I'm missing?

Thank you!

**Comment by williamboman** [comment, marked as answer]
Hello! The snippet that contains the default configuration is just copy pasted from the internal [`settings.lua`](https://github.com/williamboman/mason.nvim/blob/03a326ad32eaa227f290cad3eb562d33467af4e6/lua/mason/settings.lua#L1) file. It has an import for `path` (which is an internal, non-public, mason module). The reason it's used is for nicer cross-platform paths. If you only intend to use your neovim config on a UNIX system then you can substitute that with something like

```lua
install_root_dir = vim.fn.stdpath("data") .. "/mason"
```

**Reply by syntaxman** [reply]
Thank you for answering!

I see now what happened.  When the copy/paste was done, the module definition was appropriately left out of the copy in the README.md.  However, there is a 'require' statement on line 1, above the `local M = {}` statement:
`local path = require('mason-core.path')`
Adding that to the config file makes it work.

I am only going to use it on Linux, but understanding what was happening  makes my brain feel better :-D
